src/test.rs

```diff
@@ -73,6 +76,168 @@ pub(crate) async fn get_client_options() -> &'static ClientOptions {
         })
         .await
 }
+pub(crate) async fn auth_enabled() -> bool {
+    get_client_options().await.credential.is_some()
+}
+
+async fn get_server_version() -> &'static semver::Version {
+    static SERVER_VERSION: OnceCell<semver::Version> = OnceCell::const_new();
```

Comment:
I wonder if it might be better to bundle the `static` values into a single struct with a single lazy initializer; as it is a new `Client` has to be constructed for each group of related functions and that would collapse it down to a single one.

This is well into micro-optimization territory, though, so up to you if you want to implement that or merge as-is :)